x-pack/plugins/maps/public/ems_autosuggest/ems_autosuggest.ts

```diff
@@ -105,92 +74,87 @@ export async function suggestEMSTermJoinConfig(
   return uniqMatches.length ? uniqMatches[0].config : null;
 }
 
-function suggestByName(columnName: string): EMSTermJoinConfig[] {
-  const matches = wellKnownColumnNames.filter((wellknown) => {
-    return columnName.match(wellknown.regex);
-  });
-
-  return matches.map((m) => {
-    return m.emsConfig;
-  });
-}
+async function suggestByName(
+  columnName: string,
+  sampleValues?: SampleValues
+): Promise<EMSTermJoinConfig[]> {
+  const fileLayers = await getEmsFileLayers();
 
-function suggestByValues(values: Array<string | number>): EMSTermJoinConfig[] {
-  const matches = wellKnownColumnFormats.filter((wellknown) => {
-    for (let i = 0; i < values.length; i++) {
-      const value = values[i].toString();
-      if (!value.match(wellknown.regex)) {
-        return false;
+  const matches: EMSTermJoinConfig[] = [];
+  fileLayers.forEach((fileLayer) => {
+    const emsFields: FileLayerFieldShim[] = fileLayer.getFields();
+    emsFields.forEach((emsField: FileLayerFieldShim) => {
+      if (!emsField.alias || !emsField.alias.length) {
+        return;
       }
-    }
-    return true;
-  });
 
-  return matches.map((m) => {
-    return m.emsConfig;
+      const emsConfig = {
+        layerId: fileLayer.getId(),
+        field: emsField.id,
+      };
+      emsField.alias.forEach((alias: string) => {
+        const regex = new RegExp(alias, 'i');
+        const nameMatchesAlias = !!columnName.match(regex);
+        // Check if this violates any known id-values.
+
+        let isMatch: boolean;
+        if (sampleValues) {
+          if (emsField.values && emsField.values.length) {
+            isMatch = nameMatchesAlias && allSamplesMatch(sampleValues, emsField.values);
+          } else {
+            // requires validation against sample-values but EMS provides no meta to do so.
+            isMatch = false;
+          }
+        } else {
+          isMatch = nameMatchesAlias;
+        }
+
+        if (isMatch) {
+          matches.push(emsConfig);
+        }
+      });
+    });
   });
-}
 
-function existsInEMS(emsJson: any, emsFieldId: string, sampleValue: string): boolean {
-  for (let i = 0; i < emsJson.features.length; i++) {
-    const emsFieldValue = emsJson.features[i].properties[emsFieldId].toString();
-    if (emsFieldValue.toString() === sampleValue) {
-      return true;
-    }
-  }
-  return false;
+  return matches;
 }
 
-function matchesEmsField(emsJson: any, emsFieldId: string, sampleValues: Array<string | number>) {
+function allSamplesMatch(sampleValues: SampleValues, ids: string[]) {
   for (let j = 0; j < sampleValues.length; j++) {
     const sampleValue = sampleValues[j].toString();
-    if (!existsInEMS(emsJson, emsFieldId, sampleValue)) {
+    if (!existInIds(sampleValue, ids)) {
```

Comment:
How about just using [Array.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) instead of writing your own?